packages/framework/aqueduct/src/data-object-factories/dataObjectFactory.ts

```diff
@@ -55,6 +70,85 @@ export class DataObjectFactory<
 			mergedObjects.push(SharedMap.getFactory());
 		}
 
-		super(type, ctor, mergedObjects, optionalProviders, registryEntries, runtimeFactory);
+		let converted = false;
+		const convertRoundTripP = new Deferred<void>();
+
+		const fullConvertDataFn =
+			convertDataFn === undefined
+				? undefined
+				: async (runtime: FluidDataStoreRuntime) => {
+						if (!converted) {
+							converted = true;
```

Comment:
how do we know if data store was already converted? Maybe the snapshot we are loading from is already in new schema?